source/renderer/app/components/widgets/forms/PinCode.js

```diff
@@ -31,42 +39,258 @@ export default class PinCode extends Component<Props> {
   inputsRef = [];
   focusKey = 0;
   add = false;
+  fromBackspace = false;
+
+  state = {
+    isBackSpace: false,
+    focusKeyChanged: false,
+    focusIsUpdated: false,
+  };
+
+  hasInputNewValue = (inputNewValue: string, key: number) => {
+    const { value } = this.props;
+    const valueIsEmpty =
+      !value[key] || value[key] === '' || inputNewValue === '';
+    const valueHasChanged =
+      inputNewValue &&
+      inputNewValue.length === 1 &&
+      value[key] !== inputNewValue;
+    return valueIsEmpty || valueHasChanged;
```

Comment:
Are you sure that this is correct. Function says `hasInputNewValue` and you are returning `true` if value is empty and there is no check what value was before change

Reply:
Updated naming